chef/cookbooks/resolver/recipes/default.rb

```diff
@@ -29,6 +29,7 @@
 end
 
 dns_list << node[:dns][:nameservers]
+dns_list = dns_list.flatten.sort.uniq
```

Comment:
Even better is to append it directly a line above as the methods are not manipulating the original object.